core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -301,8 +292,9 @@ class GroupCoordinator(val brokerId: Int,
     }
 
     // store the group metadata without holding the group lock to avoid the potential
-    // for deadlock when the callback is invoked
-    delayedGroupStore.foreach(groupManager.store)
+    // for deadlock if the callback is invoked holding other locks (e.g. the replica
+    // state change lock)
+    delayedGroupStore.map(groupManager.store)
```

Comment:
Why did you change from `foreach` to `map`? It doesn't seem like the return value is being used.